integration-tests/test/theater.js

```diff
@@ -1,3 +1,49 @@
+import { theaterImageAndText } from "./lib/sources.js";
+import { THEATER } from "./lib/MiniAppType.js";
+import {
+  INITIAL_STATUS_MESSAGES,
+  EXIT_STATUS_MESSAGE,
+  assertMessagesEqual,
+  verifyMessagesReceived
+} from "./helpers/testHelpers.js";
+import { expect } from "chai";
+
 describe("Theater", () => {
-  // TODO
+  it("Runs Theater project", (done) => {
+    const expectedMessages = [
+      ...INITIAL_STATUS_MESSAGES,
+      {type: "STATUS", value: "SENDING_VIDEO", detail: {totalTime: "0"}},
+      {type: "THEATER", value: "VISUAL_URL", detail: {url: "https://javabuilder-content.code.org/abc-123/theaterImage.gif"}},
+      {type: "THEATER", value:"AUDIO_URL", detail: {url: "https://javabuilder-content.code.org/abc-123/theaterAudio.wav"}},
+      EXIT_STATUS_MESSAGE
+    ];
+
+    const assertOnMessagesReceived = receivedMessages => {
+      const verifyDetailKey = (key, receivedMessage, expectedMessage) => {
+        if (key === 'url') {
+          if (receivedMessage.value === 'VISUAL_URL') {
+            verifyMediaURL(receivedMessage.detail.url, 'theaterImage.gif');
+          } else if (receivedMessage.value === 'AUDIO_URL') {
+            verifyMediaURL(receivedMessage.detail.url, 'theaterAudio.wav');
+          } else {
+            throw new Error(`Unexpected URL received in detail for message of type ${receivedMessage.type}`);
+          }
+        } else {
+          expect(receivedMessage.detail[key]).to.equal(expectedMessage.detail[key]);
+        }
+      }
+
+      assertMessagesEqual(receivedMessages, expectedMessages, verifyDetailKey);
+    }
+
+    verifyMessagesReceived(theaterImageAndText, THEATER, assertOnMessagesReceived, done);
+  }).timeout(20000);
 });
+
+const verifyMediaURL = (url, filename) => {
+  expect(url.includes("content.code.org")).to.be.true;
```

Comment:
Wasn't sure whether to verify the entire URL (do we want integration tests to work on any javabuilder deploy, or just test?) -- can make this more substantial if we'd like.

Reply:
Yeah good question - doesn't seem super critical to me that we verify the full URL. If we wanted to, we could cross reference by pulling out the subdomain value from JAVABUILDER_WEBSOCKET_URL in environment.js, but that doesn't seem as important as verifying the filename.

Reply:
Yep, looks good!